subprojects/testing-base/src/main/java/org/gradle/api/internal/tasks/testing/worker/TestEventSerializer.java

```diff
@@ -199,6 +205,9 @@ public void write(Encoder encoder, DefaultTestFailure value) throws Exception {
             for (TestFailure cause : value.getCauses()) {
                 write(encoder, (DefaultTestFailure) cause);
             }
+            Throwable rawFailure = value.getRawFailure();
```

Comment:
🧐 If adding a `readFailure()` method (here or elsewhere), it would be good to add a `writeFailure()` method too, so that it is obvious these 2 sections of code must remain in sync.  This could be noted in the docstrings.

Reply:
Good point.  Done.